02_classify_stratified/nbconverted/plot_purity_results.py

```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# ## Plot tumor purity prediction results
+# 
+# Here, we'll visualize the results of our tumor purity prediction. For now, we're trying to predict whether a given sample is above or below the median tumor purity (i.e. classification).
+# 
+# Results analyzed here are generated by the `run_purity_prediction.py` script.
+
+# In[1]:
+
+
+from pathlib import Path
+
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+import seaborn as sns
+
+import mpmp.config as cfg
+import mpmp.utilities.analysis_utilities as au
+
+
+# In[2]:
+
+
+# set results directory
+results_dir = Path(cfg.results_dir,
+                   'purity_binarized_preds',
+                   'purity').resolve()
+
+
+# In[3]:
+
+
+results_df = (
+    au.load_purity_binarized_results(results_dir)
+      .drop(columns=['identifier'])
+)
+# all me_450k results are compressed
+results_df[results_df.training_data == 'me_450k'] = 'me_450k_compressed'
+print(results_df.shape)
+print(results_df.training_data.unique())
+results_df.head()
+
+
+# In[4]:
+
+
+sns.set({'figure.figsize': (8, 6)})
+sns.set_style('whitegrid')
+plot_df = (
+    results_df[results_df.data_type == 'test']
+      .sort_values(by=['signal', 'training_data'])
+)
+
+sns.boxplot(data=plot_df, x='signal', y='aupr', hue='training_data')
+plt.title('Binarized tumor purity prediction performance, by data type')
+plt.legend(title='Data type')
+plt.xlabel('Signal or shuffled')
+plt.ylabel('AUPR')
```

Comment:
So this is the PR. What about ROC? Does it make sense to look at that as well?

Reply:
Just curious: is this the average precision? How do you calculate the area under the PR curve?

Reply:
> So this is the PR. What about ROC? Does it make sense to look at that as well?

We've generally used PR in our past experiments since it deals better with imbalanced classes (especially when there are very few positively labeled examples; see, e.g. [this paper](https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0118432)), but in this case the labels are pretty balanced so ROC would be fine.

In any case, I did look at ROC here too, and it looks essentially the same as the PR results. I'll add those plots to this script as well.

> Just curious: is this the average precision? How do you calculate the area under the PR curve?

Yep, it's average precision, using [the sklearn implementation here](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.average_precision_score.html).